### ISSUE TITLE
[GRAPH][S1] max_batch_size above 64 silently drops every decode graph, no clamp and no log line

### ISSUE THREAD

**Issue by kekzl** [kekzl/imp#1646]
## Fact

- `runtime.max_batch_size` is accepted unclamped and unwarned, but every decode-graph path is gated on `n_sequences <= kMaxGraphPoolSize` (64), so a decode step with 65+ sequences runs entirely eager with no log line — `src/runtime/engine.h:433`, `src/runtime/engine_scheduler.cpp:1976`.

## Evidence

```
src/runtime/engine.h:433-434
    static constexpr int kMaxGraphPoolSize = 64;
    CudaGraphRunner decode_graph_pool_[kMaxGraphPoolSize];  // index = n_sequences - 1

src/runtime/engine_scheduler.cpp:1975-1977
    int graph_idx = gpu_batch.n_sequences - 1;
    if (config_.use_cuda_graphs && !profiling && gpu_batch.n_sequences > 0 && graph_idx < kMaxGraphPoolSize &&
        decode_batch_pool_.is_allocated()) {

src/runtime/engine_scheduler.cpp:2571-2574   (pipelined batched decode, same cut-off)
    const int n = static_cast<int>(rows.size());
    if (n < 2 || n > kMaxGraphPoolSize)
        return false;
```

- The `else` branch of `:1976` is `executor_->forward_logits(state, decode_logits_out, dec_stream);` (`src/runtime/engine_scheduler.cpp:2062-2065`) — a fully eager forward. No `IMP_LOG_*` on that branch.
- No clamp and no warning on the configured value:

```
src/runtime/engine_init_resolver.cpp:381-383
    } else {
        IMP_LOG_INFO("max_batch_size: %d (configured)", config_.max_batch_size);
    }
```

```
$ grep -n "max_batch_size" src/runtime/config.cpp
122:    I("runtime.max_batch_size", cfg.runtime.max_batch_size);
```

- The auto path caps at 32 (`kMaxAutoBatch`, `src/runtime/engine_init_resolver.cpp:348`), so only an explicit `--max-batch N` / `--set runtime.max_batch_size=N` with `N > 64` reaches it. The batch cap that actually binds is the configured value: `src/runtime/engine_scheduler.cpp:1520-1523` (`decode_batch.resize(max_bs)`), and `decode_batch_pool_.allocate(config_.max_batch_size, ...)` (`src/runtime/engine_kv_cache_init.cpp:748`) still allocates, so `is_allocated()` stays true and the only failing condition is the pool index.

## Impact

- Per decode step at `n_sequences > 64`: 0 graph replays, `forward_logits` issued as individual launches. The captured decode body is 44-137 kernels on Qwen3-8B (`docs/roadmap.md:183`), so the step goes from 1 `cudaGraphLaunch` to 44-137 driver launches, every step, for every request in the batch.
- Measured reference for what graph replay is worth on this tree: graphs ON/OFF tg256 = 449.04 / 188.42 tok/s = **2.38x** (`_audit` L3 baseline, batch 1). The batch-65 magnitude is not that number and is not measured — see Unverified.
- Blast radius: any operator who configures concurrency above 64. `--max-batch 64` is already a configuration this repo reasons about (`AUDIT.md` C6 sizes `spec_graphs_` "~1950 at `--max-batch 64`"), so 65+ is one step past a documented setting.
- Silent: the operator sees `max_batch_size: 128 (configured)` at init and nothing else. `Resolved dispatch: … graphs=1` still prints, because `config_.use_cuda_graphs` is true — graphs are enabled and never used.

## Repro

NOT RUN (measurement-only, needs the GPU):

```
docker compose run --rm imp-server --model /models/Qwen3-8B-Q8_0.gguf \
  --set runtime.max_batch_size=80 --set runtime.max_seq_len=4096
# drive 80 concurrent streaming completions, then:
grep -c "capturing CUDA graph" server.log     # expect: captures only for n<=64
```

Static, runnable now:

```
$ grep -n "kMaxGraphPoolSize" src/runtime/engine.h src/runtime/engine_scheduler.cpp
$ grep -n "max_batch_size" src/runtime/config.cpp
```

## Fix direction

- Clamp `config_.max_batch_size` to `kMaxGraphPoolSize` in the resolver, or refuse the value at parse time the way an unknown `--set` key is refused (`#1627` precedent).
- If large batches are wanted, key `decode_graph_pool_` by a bucketed batch size (pow2, as `bucket_pow2` already does for `max_blocks` and `max_ctx` at `engine_scheduler.cpp:1990-2021`) instead of an exact index.
- Minimum: one WARN at init naming the graph cut-off, and one on the first decode step that falls through the `:1976` gate.

## Acceptance criteria

1. A server started with `--set runtime.max_batch_size=80` either logs a WARN naming 64 as the decode-graph ceiling, or reports `max_batch_size: 64 (clamped from 80: decode graph pool)`.
2. With the clamp or the bucketing in place, a 80-concurrent run logs `capturing CUDA graph` at least once for a step whose `n_sequences > 64`, or logs the WARN exactly once.
3. `make verify-fast` decode tg128 within 1 % of the 287.19 pin (the change must not touch the batch<=64 path).

## Unverified

- The throughput loss at `n_sequences > 64` is not measured. The 2.38x graphs-ON/OFF figure is a batch-1 measurement and does not transfer: at batch 65 the per-launch host cost is amortised over 65 rows of work, so the loss is expected to be smaller than 2.38x and is not otherwise bounded by anything in this repo.

**Comment by kekzl** [comment]
Fixed in 836b6f93 (#1709). Verified on main.
